### PR TITLE
API-58 Add transactional wrapper with automatic commit/rollback

### PR DESCRIPTION
**Pull request by avardugin**

### CODE REVIEW

File: lib/transaction.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { AggregateError } from 'selvera-aggregate-error';
  */
 export class Transaction {
     private isCompleted = false;
-    private constructor(private readonly client: Client) { }
+    private constructor(public readonly client: Client) { }
```
Comment (pcwiek) [inline lib/transaction.ts]: Let's leave it `private` 😉

File: lib/transaction.ts [hunk 2]
```diff
@@ -99,3 +99,22 @@ export class Transaction {
         return new Transaction(client);
     }
 }
+
+export async function executeTransactionally<T>(
+    db: Pool,
+    fn: (transaction: Client) => Promise<T>
```
Comment (pcwiek) [inline lib/transaction.ts]: If we type it as `fn: (transaction: Transaction) => Promise<T>` we won't have to expose the wrapped client

File: lib/transaction.ts [hunk 3]
```diff
@@ -99,3 +99,22 @@ export class Transaction {
...
+): Promise<T> {
+    const transaction = await Transaction.begin(db);
+    try {
+        const result = await fn(transaction.client);
```
Comment (pcwiek) [inline lib/transaction.ts]: ... and we could just pass `fn(transaction)` here

**Comment by pcwiek** [comment]
Those changes will probably impact tests, so I didn't really review them just yet.

**Comment by GrzZebrowski** [comment]
@pcwiek maybe simply `executeInTransaction()`? `executeTransactionally()` also sounds ok.

**Comment by pcwiek** [comment]
@jonganc we should remove the glob altogether. TSLint will read `tsconfig.json` by default. But that's a small task for another PR

**Comment by pcwiek** [comment]
Since there's no strong preference, I'll just merge this PR, since `executeTransactionally` reads just fine

**Pull request merged**